apps/hash-graph/lib/graph/src/store/fetcher.rs

```diff
@@ -131,3 +109,213 @@ where
         &mut self.store
     }
 }
+
+#[async_trait]
+impl<S, A, R: Record> Read<R> for FetchingStore<S, A>
+where
+    A: Send + Sync,
+    S: Read<R> + Send,
+{
+    async fn read(
+        &self,
+        query: &Filter<R>,
+        time_projection: &TimeProjection,
+    ) -> Result<Vec<R>, QueryError> {
+        self.store.read(query, time_projection).await
+    }
+}
+
+#[async_trait]
+impl<S, A> AccountStore for FetchingStore<S, A>
+where
+    S: AccountStore + Send,
+    A: Send + Sync,
+{
+    async fn insert_account_id(&mut self, account_id: AccountId) -> Result<(), InsertionError> {
+        self.store.insert_account_id(account_id).await
+    }
+}
+
+#[async_trait]
+impl<S, A> DataTypeStore for FetchingStore<S, A>
```

Comment:
A little bit verbose/annoying we have to impl everything and can't rely on fallbacks. You've already mentioned before you pushed for a modular approach, unfortunate that we didn't end up with that :/

Reply:
Yeah, it's a little unfortunate, but we need to adjust both, the creation and the updating logic, so it's not that bad. Only the `AccoutStore` is completely untouched.